pkg/sessionctx/variable/sysvar.go

```diff
@@ -899,11 +899,6 @@ var defaultSysVars = []*SysVar{
 		}
 		return setTiDBTableValue(s, "tikv_gc_concurrency", val, "How many goroutines used to do GC parallel, [1, 256], default 2")
 	}},
-	{Scope: ScopeGlobal, Name: TiDBGCScanLockMode, Value: "LEGACY", Type: TypeEnum, PossibleValues: []string{"PHYSICAL", "LEGACY"}, GetGlobal: func(_ context.Context, s *SessionVars) (string, error) {
```

Comment:
Hello, I have tried to connect to the tidb with current version to a existing TIDB cluster, the compatibility test results are as follows, and I think they meet expectations:
```
mysql> select * from mysql.tidb where variable_name like "%gc%";
+--------------------------+------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+
| VARIABLE_NAME            | VARIABLE_VALUE                                                                                             | COMMENT                                                                                     |
+--------------------------+------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+
| tikv_gc_leader_uuid      | 63175d85bd40005                                                                                            | Current GC worker leader UUID. (DO NOT EDIT)                                                |
| tikv_gc_leader_desc      | host:wuxueliandeMacBook-Pro.local, pid:30947, start at 2023-12-12 17:57:49.384276 +0800 CST m=+2.872929084 | Host name and pid of current GC leader. (DO NOT EDIT)                                       |
| tikv_gc_leader_lease     | 20231212-18:03:49.905 +0800                                                                                | Current GC worker leader lease. (DO NOT EDIT)                                               |
| tikv_gc_auto_concurrency | true                                                                                                       | Let TiDB pick the concurrency automatically. If set false, tikv_gc_concurrency will be used |
| tikv_gc_enable           | true                                                                                                       | Current GC enable status                                                                    |
| tikv_gc_run_interval     | 10m0s                                                                                                      | GC run interval, at least 10m, in Go format.                                                |
| tikv_gc_life_time        | 10m0s                                                                                                      | All versions within life time will not be collected by GC, at least 10m, in Go format.      |
| tikv_gc_last_run_time    | 20231212-17:57:49.679 +0800                                                                                | The time when last GC starts. (DO NOT EDIT)                                                 |
| tikv_gc_safe_point       | 20231212-17:47:49.679 +0800                                                                                | All versions after safe point can be accessed. (DO NOT EDIT)                                |
| tikv_gc_mode             | distributed                                                                                                | Mode of GC, "central" or "distributed"                                                      |
| tikv_gc_scan_lock_mode   | legacy                                                                                                     | Mode of scanning locks, "physical" or "legacy"                                              |
+--------------------------+------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+
11 rows in set (0.16 sec)

mysql> select @@tidb_gc_scan_lock_mode;
ERROR 1193 (HY000): Unknown system variable 'tidb_gc_scan_lock_mode'

```
Meanwhile, I will update the doc later https://docs.pingcap.com/tidb/stable/system-variables#tidb_gc_scan_lock_mode-new-in-v50

Reply:
I can accept it if you think deleting the variable is better. What I meant before is that, perhaps we can consider keeping the variable instead of deleting it, to prevent potential failures of existing usages of it. Also it might be hard to say whether we might introduce other way to resolve locks which may need new options for this variable.
How do you think?

Reply:
And for an new cluster start with tiup, 

```tiup playground --tag master  --db 0 --pd 3 --kv 3 --monitor --host=0.0.0.0 &```
connect to the tidb with current version to the above cluster,
the result is here
```

mysql> select * from mysql.tidb where variable_name like "%gc%";
+--------------------------+-------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+
| VARIABLE_NAME            | VARIABLE_VALUE                                                                                              | COMMENT                                                                                     |
+--------------------------+-------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+
| tikv_gc_leader_uuid      | 631761bf0540002                                                                                             | Current GC worker leader UUID. (DO NOT EDIT)                                                |
| tikv_gc_leader_desc      | host:wuxueliandeMacBook-Pro.local, pid:34686, start at 2023-12-12 18:16:16.556451 +0800 CST m=+46.781755001 | Host name and pid of current GC leader. (DO NOT EDIT)                                       |
| tikv_gc_leader_lease     | 20231212-18:18:16.677 +0800                                                                                 | Current GC worker leader lease. (DO NOT EDIT)                                               |
| tikv_gc_auto_concurrency | true                                                                                                        | Let TiDB pick the concurrency automatically. If set false, tikv_gc_concurrency will be used |
| tikv_gc_enable           | true                                                                                                        | Current GC enable status                                                                    |
| tikv_gc_run_interval     | 10m0s                                                                                                       | GC run interval, at least 10m, in Go format.                                                |
| tikv_gc_life_time        | 10m0s                                                                                                       | All versions within life time will not be collected by GC, at least 10m, in Go format.      |
| tikv_gc_last_run_time    | 20231212-18:16:16.861 +0800                                                                                 | The time when last GC starts. (DO NOT EDIT)                                                 |
| tikv_gc_safe_point       | 20231212-18:06:16.861 +0800                                                                                 | All versions after safe point can be accessed. (DO NOT EDIT)                                |
+--------------------------+-------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------+
9 rows in set (0.01 sec)

mysql> select @@tidb_gc_scan_lock_mode;
ERROR 1193 (HY000): Unknown system variable 'tidb_gc_scan_lock_mode'
```